epuck_ros2_cpp/src/driver.cpp

```diff
@@ -0,0 +1,447 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+extern "C" {
+#include <assert.h>
+#include <fcntl.h>
+#include <linux/i2c-dev.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+}
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "epuck_ros2_cpp/i2c_wrapper.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+#include "geometry_msgs/msg/transform_stamped.hpp"
+#include "geometry_msgs/msg/twist.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include "sensor_msgs/msg/range.hpp"
+#include "std_msgs/msg/string.hpp"
+#include "tf2_ros/static_transform_broadcaster.h"
+#include "tf2_ros/transform_broadcaster.h"
+
+#define MSG_ACTUATORS_SIZE 20
+#define MSG_SENSORS_SIZE 47
+#define PERIOD_MS 64
+#define PERIOD_S (PERIOD_MS / 1000.0)
+#define OUT_OF_RANGE 0
+#define ENCODER_RESOLUTION 1000.0
+#define ODOM_OVERFLOW_GRACE_TICKS 2000
+
+#define MIN(X, Y) (((X) < (Y)) ? (X) : (Y))
+#define MAX(X, Y) (((X) > (Y)) ? (X) : (Y))
+#define CLIP(VAL, MIN_VAL, MAX_VAL) MAX(MIN((MAX_VAL), (VAL)), (MIN_VAL))
+#define POW2(N) (1 << (N))
+
+const float DEFAULT_WHEEL_DISTANCE = 0.05685;
+const float DEFAULT_WHEEL_RADIUS = 0.02;
+const float SENSOR_DIST_FROM_CENTER = 0.035;
+const std::vector<std::vector<float>> INFRARED_TABLE = {{0, 4095},       {0.005, 2133.33}, {0.01, 1465.73},
+                                                        {0.015, 601.46}, {0.02, 383.84},   {0.03, 234.93},
+                                                        {0.04, 158.03},  {0.05, 120},      {0.06, 104.09}};
+const std::vector<double> DISTANCE_SENSOR_ANGLE = {
+  -15 * M_PI / 180,   // ps0
+  -45 * M_PI / 180,   // ps1
+  -90 * M_PI / 180,   // ps2
+  -150 * M_PI / 180,  // ps3
+  150 * M_PI / 180,   // ps4
+  90 * M_PI / 180,    // ps5
+  45 * M_PI / 180,    // ps6
+  15 * M_PI / 180,    // ps7
+  0 * M_PI / 180      // tof
+};
+
+class EPuckPublisher : public rclcpp::Node {
+public:
+  EPuckPublisher(int argc, char *argv[]) : Node("pipuck_driver") {
+    // Parse arguments
+    std::string type = "hw";
+    for (int i = 1; i < argc; i++)
+      if (strcmp(argv[i], "--type") == 0) {
+        i++;
+        type = argv[i];
+      }
+
+    wheel_distance = declare_parameter<float>("wheel_distance", DEFAULT_WHEEL_DISTANCE);
+    wheel_radius = declare_parameter<float>("wheel_radius", DEFAULT_WHEEL_RADIUS);
+    /*
+    callback_handler =
+      this->add_on_set_parameters_callback(std::bind(&EPuckPublisher::param_change_callback, this,
+        std::placeholders::_1));
+    */
```

Comment:
Is it normal that this is commented?

Furthermore, we use `/*` comments only for temporary debug:
https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs385-use--for-all-comments-including-multi-line-comments

Reply:
You right, it is leftover from tests :/